src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala

```diff
@@ -0,0 +1,194 @@
+    val featureServiceHost: String = envOrFail(Constants.Env.FeatureServiceHost)
+    val oxfordLanguageToken: String = envOrFail(Constants.Env.OxfordLanguageToken)
+    val oxfordVisionToken: String = envOrFail(Constants.Env.OxfordVisionToken)
+
+    val appInsightsKey: Option[String] = envOrNone(Constants.Env.AppInsightsKey)
+    val modelsDir: Option[String] = envOrNone(Constants.Env.LanguageModelDir)
+
+    private def envOrFail(name: String): String = {
+      envOrNone(name) match {
+        case Some(v) => v
+        case None =>
+          DriverLog.error(s"Environment variable not defined: $name")
+          sys.exit(1)
+      }
+    }
+  }
+
+  // TODO: logging configuration should be done in log4j config file
+  AppInsights.init(Settings.appInsightsKey)
+
+  Logger.getLogger("org").setLevel(Level.ERROR)
+  Logger.getLogger("akka").setLevel(Level.ERROR)
+  Logger.getLogger("libinstagram").setLevel(Level.DEBUG)
+  Logger.getLogger("libfacebook").setLevel(Level.DEBUG)
+  Logger.getLogger("liblocations").setLevel(Level.DEBUG)
+
+  private implicit object TransformContext extends TransformContext {
+    val geofence = Geofence(north = 49.6185146245, west = -124.9578052195, south = 46.8691952854, east = -121.0945042053)
+    val placeRecognizer = new PlaceRecognizer(Settings.modelsDir)
+    val featureServiceClient = new FeatureServiceClient(Settings.featureServiceHost)
+    val locationsExtractor = new LocationsExtractor(featureServiceClient, geofence, Some(placeRecognizer)).buildLookup()
+    val keywordExtractor = new KeywordExtractor(List("Ariana"))
+    val imageAnalyzer = new ImageAnalyzer(ImageAnalysisAuth(Settings.oxfordVisionToken), featureServiceClient)
+    val languageDetector = new LanguageDetector(LanguageDetectorAuth(Settings.oxfordLanguageToken))
+    val sentimentDetector = new SentimentDetector(SentimentDetectorAuth(Settings.oxfordLanguageToken))
+    val supportedLanguages = Set("en", "fr", "de")
+  }
+
+  private def createStreamingContext(): StreamingContext = {
+    val conf = new SparkConf()
+      .setAppName(Constants.SparkAppName)
+      .setIfMissing("spark.master", Constants.SparkMasterDefault)
+
+    implicit val streamingContext = new StreamingContext(
+      new SparkContext(conf),
+      Seconds(Constants.SparkStreamingBatchSizeDefault))
+
+    val streamProvider = StreamProviderFactory.create()
+    val streamRegistry = getStreamRegistry
+
+    // Attach each pipeline (aka code path)
+    // 'fortisEvents' is the stream of analyzed data aggregated (union) from all pipelines
+    val fortisEvents = pipelines.flatMap(
+      pipeline => pipeline(streamProvider, streamRegistry)
+    ).reduceOption(_.union(_))
+
+    // TODO: other computations and save to DB
+    fortisEvents.foreach(
+      _.foreachRDD(
+        _.foreachPartition(throw new NotImplementedException)))
+
+    streamingContext
+  }
+
+  // Main starts here
+  val ssc = StreamingContext.getOrCreate(Settings.progressDir, createStreamingContext)
```

Comment:
I take it this is the call that [enables checkpointing](https://spark.apache.org/docs/1.6.1/api/java/org/apache/spark/streaming/StreamingContext.html#getOrCreate(java.lang.String,%20scala.Function0,%20org.apache.hadoop.conf.Configuration,%20boolean)). I was expecting a call to [SparkContext.setCheckpointDir(String)](https://jaceklaskowski.gitbooks.io/mastering-apache-spark/spark-rdd-checkpointing.html) or [StreamingContext.checkpoint(String)](https://spark.apache.org/docs/latest/streaming-programming-guide.html#how-to-configure-checkpointing) which would be more explicit by making it obvious what's going on without having to know the details of the Spark API. See also the [principle of least surprise](http://wiki.c2.com/?PrincipleOfLeastAstonishment).